trackintel/geogr/distances.py

```diff
@@ -82,44 +83,41 @@ def point_haversine_dist(lon_1, lat_1, lon_2, lat_2, r=6371000, float_flag=False
     return r * np.arccos(cos_lat_d - cos_lat1 * cos_lat2 * (1 - cos_lon_d))
 
 
-def calculate_distance_matrix(X, Y=None, dist_metric="haversine", n_jobs=0, **kwds):
+def calculate_distance_matrix(X, Y=None, dist_metric="haversine", n_jobs=None, **kwds):
     """
-    Calculate a distance matrix based on a specific distance metric.
+    Compute the distance matrix from a vector array X and optional Y.
 
+    X and Y can either be Point geometries or LineString geometries.
     If only X is given, the pair-wise distances between all elements in X are calculated.
     If X and Y are given, the distances between all combinations of X and Y are calculated.
-    Distances between elements of X and X, and distances between elements of Y and Y are not calculated.
 
     Parameters
     ----------
-    X : Trackintel class
+    X : GeoDataFrame
 
-    Y : Trackintel class, optional
-        Should be of the same type as X
+    Y : GeoDataFrame, optional
+        Should have same geometry type as X
 
     dist_metric: {{'haversine', 'euclidean', 'dtw', 'frechet'}}, optional
         The distance metric to be used for calculating the matrix. By default 'haversine.
 
-        For staypoints or positionfixes, a common choice is 'haversine' or 'euclidean'. This function wraps around
-        the ``pairwise_distance`` function from scikit-learn if only `X` is given and wraps around the
-        ``scipy.spatial.distance.cdist`` function if X and Y are given.
+        For Point geometries we provide the 'haversine' metric.
+        This function then wraps `sklearn.metrics.pairwise_distances`.
         Therefore the following metrics are also accepted:
 
-        via ``scikit-learn``: `['cityblock', 'cosine', 'euclidean', 'l1', 'l2', 'manhattan']`
+        - via ``scikit-learn``: `['cityblock', 'cosine', 'euclidean', 'l1', 'l2', 'manhattan']`
 
-        via ``scipy.spatial.distance``: `['braycurtis', 'canberra', 'chebyshev', 'correlation', 'dice', 'hamming', 'jaccard',
+        - via ``scipy.spatial.distance``: `['braycurtis', 'canberra', 'chebyshev', 'correlation', 'dice', 'hamming', 'jaccard',
         'kulsinski', 'mahalanobis', 'minkowski', 'rogerstanimoto', 'russellrao', 'seuclidean', 'sokalmichener',
         'sokalsneath', 'sqeuclidean', 'yule']`
 
-        For triplegs, common choice is 'dtw' or 'frechet'. This function uses the implementation
-        from similaritymeasures.
+        For LineStrings, we provide the metrics {'dtw', 'frechet'} via the implementation from similaritymeasures.
 
     n_jobs: int, optional
-        Number of cores to use: 'dtw', 'frechet' and all distance metrics from `pairwise_distance` (only available
-        if only X is given) are parallelized. By default 1.
+        Number of cores to use. Only used for Point-distances. Default uses all possible cores
```

Comment:
`n_jobs=None `(default) means n_jobs=1 and not `all possible cores` (?), see [here](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise_distances.html)

Reply:
Ah, yes you are right. I changed it